File: sdks/java/basys.sdk/backends/org/eclipse/basyx/aas/backend/connected/ConnectedElement.java
package org.eclipse.basyx.aas.backend.connected;

import org.eclipse.basyx.vab.core.proxy.VABElementProxy;

public class ConnectedElement {
	private VABElementProxy proxy;
	private String path;

	public VABElementProxy getProxy() {
		return proxy;
	}

	public ConnectedElement(String path, VABElementProxy proxy) {
		super();
		this.path = path;
		this.proxy = proxy;
	}

	protected String constructPath(String relativePath) {
		if (!path.isEmpty()) {
			return path + "/" + relativePath;
		} else {
			return relativePath;
		}
	}

	protected String getPath() {
		return path;
	}
}

File: sdks/c++/basys.sdk.cc/src/basyx/aas/ConnectedElement.cpp
/*
 * ConnectedElement.cpp
 *
 *      Author: wendel
 */

#include "ConnectedElement.h"

ConnectedElement::ConnectedElement(const std::string & path, const std::shared_ptr<VABElementProxy> & proxy) :
  proxy(proxy),
  path(path)
{

}

std::shared_ptr<VABElementProxy> ConnectedElement::getProxy() const
{
  return this->proxy;
}

